packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -271,11 +291,11 @@ export class Nodegroup extends Resource implements INodegroup {
       nodegroupName: props.nodegroupName,
       nodeRole: this.role.roleArn,
       subnets: this.cluster.vpc.selectSubnets(props.subnets).subnetIds,
-      amiType: props.amiType ?? (props.instanceType ? getAmiTypeForInstanceType(props.instanceType).toString() :
+      amiType: props.amiType ?? (props.instanceType ? getAmiTypeForInstanceType(props.instanceType[0]).toString() :
```

Comment:
What if the array contains different types? is choosing `0` the right thing to do?

Reply:
Addressed. Now we are using a function to determine the `amiType`.